frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -85,8 +100,106 @@ const CatalogView = (props: CatalogViewProps) => {
+				// 	dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+				// 	});	
+			} else {
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+			}
+		}
+	}, [course?.courseNumber]);
+
+
+	const gradesData = useSelector((state: State) => state.grade?.gradesData ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollmentData = useSelector((state: State) => state.enrollment?.enrollmentData ?? null);
+	
+	const gradesGraphData = useSelector((state: State) => state.grade.graphData ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollGraphData = useSelector((state: State) => state.enrollment.graphData ?? null);
+	
+	const gradesSelectedCourses = useSelector((state: State) => state.grade.selectedCourses ?? null);
+	
+	const enrollSelectedCourses = useSelector((state: State) => state.enrollment.selectedCourses ?? null);
+	
+	//may need in the future if graphs are rewritten
+	const enrollContext = useSelector((state: State) => state.enrollment.context.courses ?? null);
+
+	useEffect(() => {
+		console.log(courseNumber)
+		if (course !== null) {
+			if (legacyId !== null) {	
+				setTab(0)	
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+				
+				if (legacyId !== null) {
+					let temp = semester.split(' ');
+					dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+					axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+						dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+						});
+				}
+			} else {
+				setTab(0)	
+				dispatch(gradeReset())
+				dispatch(enrollReset())
+				dispatch(fetchEnrollContext());
+				dispatch(fetchGradeContext());
+
+				let legacyId = (store.getState() as any).enrollment?.context?.courses?.find(
+					(c: any) => c.abbreviation === abbreviation && c.course_number === courseNumber
+				)?.id ?? null
+				
+				if (legacyId !== null) {
+					let temp = semester.split(' ');
+					dispatch(fetchGradeFromUrl(`/grades/0-${legacyId}-all-all`));
+					axios.get(`/api/enrollment/sections/${legacyId}/`).then((res) => {
+						dispatch(fetchEnrollFromUrl(`/enrollment/0-${legacyId}-${temp[0]}-${temp[1]}-${res.data[0].sections[0].section_id}`))
+						});
+				}
+			}
+		}
+	}, [course?.courseNumber]);
+	
 
-	const [playlists, sections] = useMemo(() => {
+	useEffect(() => {
+		//console.log(gradesSelectedCourses)
+		if (course !== null) {
+			if (gradesSelectedCourses?.length == 1) {
+				dispatch(fetchGradeData(gradesSelectedCourses));
+			}
+		}
+	}, [gradesSelectedCourses])
+
+	useEffect(() => {
+		if (course !== null) {
+			if (enrollSelectedCourses?.length == 1) {
+				dispatch(fetchEnrollData(enrollSelectedCourses));
+			}
+		}
+	}, [enrollSelectedCourses])
```

Comment:
For these two `useEffect` on line 184, and 193, think about what events really trigger these and how you can refactor to remove them entirely by placing the dispatches inside of the event triggers themselves, instead of "listening" for them to occur.

Reply:
Fixed